achinya - Action and Func Delegates

achinya opened the pull request:
pull request with all project and test files

achinya commented:
Edited the solution file to include the change in path

achinya commented:
Pull request  for evaluation - Action and Func Delegates

CodeMazeBlog commented:
May I ask what happened here with all the commits and file uploads?

achinya commented:
When i read through the full log for Jenkins build, there was a slightly different path for test folder and its proj files, so had to delete and upload few files again. Also the something with nuget reference so did a restore and uploaded the files.

Regards

Aparna

> On 17-Dec-2021, at 10:41 PM, Code Maze ***@***.***> wrote:
> 
> ﻿
> May I ask what happened here with all the commits and file uploads?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you modified the open/close state.

CodeMazeBlog reviewed (commented):
Generally, the code is alright. There's a lot of small corrections I would make like redundant empty lines, missing newlines.... And you should definitely let the IDE do the formatting for you.

CodeMazeBlog commented on csharp-advanced-topics/Dele_Types_Action_Func/Tests/DelegatesTest.cs:
```diff
@@ -0,0 +1,67 @@
+        [TestMethod]
+        public void ForActionDelegate_whenTwoParametersAreSent()
+        {
+            int x = 20;
+            int y = 5;
```
We should use var whenever we can.

CodeMazeBlog commented on csharp-advanced-topics/Dele_Types_Action_Func/Tests/DelegatesTest.cs:
```diff
@@ -0,0 +1,67 @@
+        [TestMethod]
+        public static void ForFuncDelegate_whenTwoParametersAreSent()
+        {
+            int a = 10;
+            int b = 20;
```
We should use var whenever we can.

CodeMazeBlog commented on csharp-advanced-topics/Dele_Types_Action_Func/Tests/Delegates_Test.csproj:
```diff
@@ -0,0 +1,31 @@
+    <ProjectReference Include="..\..\Delegates_Action_Func\Delegates_Action_Func.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="Delegates_Action_Func">
+      <HintPath>..\..\Delegates_Action_Func\bin\Debug\net6.0\Delegates_Action_Func.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.VisualStudio.TestPlatform.TestFramework">
+      <HintPath>..\..\Delegates_Action_Func\bin\Debug\net6.0\Microsoft.VisualStudio.TestPlatform.TestFramework.dll</HintPath>
+    </Reference>
+  </ItemGroup>
```
This seems strange, what's happening here, why is it referencing DLLs from the output folder?

achinya replied:
I am not exactly sure why that reference is coming there. I have not added any extra references from outside. The only thing I had to do was to delete and add the test framework NuGet packages again since it did not take the older existing version. Maybe this reference was added after that. As I have mentioned in my emails before the PR,there was a problem with the NuGet manager for VS2022 which was resolved through multiple restores.

CodeMazeBlog commented on csharp-advanced-topics/Dele_Types_Action_Func/Tests/DelegatesTest.cs:
```diff
@@ -0,0 +1,67 @@
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Delegates_Test
+{
```
Namespaces shouldn't have underscores in the name.

CodeMazeBlog commented on csharp-advanced-topics/Dele_Types_Action_Func/Delegates_Action_Func/Program.cs:
```diff
@@ -0,0 +1,46 @@
+﻿// See https://aka.ms/new-console-template for more information
+namespace Delegates_Action_Func
```
Namespaces shouldn't have underscores.

achinya commented:
Hello,

I have done the above changes in the code files. The formatting is done in
the IDE only and i have removed the extra lines and spaces.
However for the below suggestion -

In csharp-advanced-topics/Dele_Types_Action_Func/Tests/Delegates_Test.csproj
<https://github.com/CodeMazeBlog/CodeMazeGuides/pull/110#discussion_r772575711>
:

> +  <ItemGroup>
+    <Reference Include="Delegates_Action_Func">
+      <HintPath>..\..\Delegates_Action_Func\bin\Debug\net6.0\Delegates_Action_Func.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.VisualStudio.TestPlatform.TestFramework">
+      <HintPath>..\..\Delegates_Action_Func\bin\Debug\net6.0\Microsoft.VisualStudio.TestPlatform.TestFramework.dll</HintPath>
+    </Reference>
+  </ItemGroup>

This seems strange, what's happening here, why is it referencing DLLs from
the output folder?

I am also not exactly sure why that reference is coming there. I have not
added any extra references from outside. The only thing I had to do was to
delete and add the test framework NuGet packages again since it did not
take the older existing version. Maybe this reference was added after that.
As I have mentioned in my emails before the PR, there was a problem with
the NuGet manager for VS2022 which was resolved through multiple restores.
I am not sure what I have to correct here as I do not want the builds to
fail because of references.

I have made the changes in the respected files. As for the multiple
commits, I am sorry for this as I had written and uploaded the code in the
old version of .net. After that, I switched to .Net 6.0 and wrote the whole
code again. I hope to do this properly with fewer commits going further.

Thank you

Regards,

Aparna

On Mon, Dec 20, 2021 at 11:45 PM Code Maze ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> csharp-advanced-topics/Dele_Types_Action_Func/Delegates_Action_Func/Program.cs
> <https://github.com/CodeMazeBlog/CodeMazeGuides/pull/110#discussion_r772576143>
> :
>
> > @@ -0,0 +1,46 @@
>
> +﻿// See https://aka.ms/new-console-template for more information
>
> +namespace Delegates_Action_Func
>
>
> Namespaces shouldn't have underscores.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeMazeBlog/CodeMazeGuides/pull/110#pullrequestreview-836687857>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AW4DF6EHQYD36CFWNDXWS7TUR5XEHANCNFSM5KIGYIMQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
